testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -518,6 +518,62 @@ public void testSelectInTransaction()
         });
     }
 
+    @Test
+    public void testSelectVersionOfNonExistentTable()
+    {
+        String catalog = getSession().getCatalog().orElseThrow();
+        String schema = getSession().getSchema().orElseThrow();
+        String tableName = "foo_" + randomTableSuffix();
+        assertThatThrownBy(() -> query("SELECT * FROM " + tableName + " FOR TIMESTAMP AS OF TIMESTAMP '2021-03-01 00:00:01'"))
+                .hasMessage(format("line 1:15: Table '%s.%s.%s' does not exist", catalog, schema, tableName));
+        assertThatThrownBy(() -> query("SELECT * FROM " + tableName + " FOR VERSION AS OF 'version1'"))
+                .hasMessage(format("line 1:15: Table '%s.%s.%s' does not exist", catalog, schema, tableName));
+    }
+
+    /**
+     * A connector can support FOR TIMESTAMP, FOR VERSION, both or none. With FOR TIMESTAMP/VERSION is can support some types but not the others.
+     * Because of version support being multidimensional, {@link TestingConnectorBehavior} is not defined. The test verifies that query doesn't fail in
+     * some weird way, serving as a smoke test for versioning. The purpose of the test is to validate the connector does proper validation.
+     */
+    @Test
+    public void testTrySelectTableVersion()
```

Comment:
I am not sure I get the intent here. Is there expectation from specific connector to provide more verbose version of this test which actually verifies specifically which query should fail and which should not?

Reply:
My expectation is that a connector supporting time travel will have its own test covering it.
Here, we only cover cases and verify they don't end up with some NPE, IAE, ISE or something.